src/components/execution/AddExecutionForm.js

```diff
@@ -0,0 +1,288 @@
+    textAlign: 'center',
+    marginTop: theme.spacing(2),
+  },
+  buttonWrapper: {
+    display: 'inline-block',
+  },
+  menuItem: {
+    padding: theme.spacing(0.5, 4),
+  },
+});
+
+class AddExecutionForm extends Component {
+  static propTypes = {
+    results: PropTypes.instanceOf(List),
+    datasets: PropTypes.instanceOf(List),
+    algorithms: PropTypes.instanceOf(List),
+    t: PropTypes.func.isRequired,
+    dispatchGetDatasets: PropTypes.func.isRequired,
+    dispatchGetAlgorithms: PropTypes.func.isRequired,
+    dispatchExecuteAlgorithm: PropTypes.func.isRequired,
+    dispatchGetResults: PropTypes.func.isRequired,
+    classes: PropTypes.shape({
+      formControl: PropTypes.string.isRequired,
+      infoAlert: PropTypes.string.isRequired,
+      buttonContainer: PropTypes.string.isRequired,
+      buttonWrapper: PropTypes.string.isRequired,
+      menuItem: PropTypes.string.isRequired,
+    }).isRequired,
+    isLoading: PropTypes.bool.isRequired,
+    pythonVersion: PropTypes.shape({
+      valid: PropTypes.bool,
+      version: PropTypes.string,
+    }).isRequired,
+  };
+
+  static defaultProps = {
+    datasets: null,
+    algorithms: null,
+    results: null,
+  };
+
+  state = {
+    sourceId: '',
+    algorithmId: '',
+    filename: '',
+  };
+
+  componentDidMount() {
+    const {
+      dispatchGetDatasets,
+      dispatchGetAlgorithms,
+      dispatchGetResults,
+    } = this.props;
+    dispatchGetDatasets();
+    dispatchGetAlgorithms();
+    dispatchGetResults();
+  }
+
+  handleDatasetSelectOnChange = (e) => {
+    this.setState({ sourceId: e.target.value });
+  };
+
+  handleAlgorithmSelectOnChange = (e) => {
+    this.setState({ algorithmId: e.target.value });
+  };
+
+  handleNameOnChange = (e) => {
+    this.setState({ filename: e.target.value });
+  };
+
+  executeAlgorithm = () => {
+    const { dispatchExecuteAlgorithm, algorithms } = this.props;
+    const { sourceId, algorithmId, filename } = this.state;
+    const { language } = algorithms.find(({ id }) => id === algorithmId);
+    if (language) {
+      dispatchExecuteAlgorithm({ sourceId, algorithmId, language, filename });
+    }
+  };
+
+  renderDatasetsAndResultsSelect = () => {
+    const { sourceId } = this.state;
+    const { classes, t, datasets, results } = this.props;
+
+    if (datasets?.isEmpty() || results?.isEmpty()) {
+      return <Loader />;
+    }
+
+    const datasetMenuItems = datasets
+      .sortBy(({ name }) => name)
+      .map(({ id, name }) => (
+        <MenuItem value={id} key={id} className={classes.menuItem}>
+          {name}
+        </MenuItem>
+      ));
+
+    const resultMenuItems = results
+      .sortBy(({ name }) => name)
+      .map(({ id, name }) => (
+        <MenuItem value={id} key={id} className={classes.menuItem}>
+          {name}
+        </MenuItem>
+      ));
+
+    return (
+      <FormControl variant="outlined" className={classes.formControl}>
+        <InputLabel id="dataset-select">
```

Comment:
so the `<InputLabel>`s have this ugly behavior:

![Screen Shot 2020-11-09 at 11 17 40 AM](https://user-images.githubusercontent.com/19311953/98529052-72f75400-227d-11eb-88b9-f460d0013abb.png)

a quick fix would be to add a `backgroundColor: '#F7F7F7'` to them (same as the background color of `BrowserWindow`):

![Screen Shot 2020-11-09 at 11 18 33 AM](https://user-images.githubusercontent.com/19311953/98529107-899dab00-227d-11eb-9518-b1ea20f6b68c.png)

Reply:
I agree, I will do as you mentioned.

Reply:
As mentioned in my last review, This is caused by the `label` prop value in the select below not matching the one of the `InputLabel`